### ISSUE TITLE
[HOLD for payment 2024-11-11] Web - Search - Console error shows up when entering comma in the search router

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#51661]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.55-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+kh1610015@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com
2. Click Search router icon
3. Enter a comma

## Expected Result:
No console error when entering comma in the search router

## Actual Result:
Console error shows up when entering comma in the search router

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/9d8fa06a-b9cf-42a1-88ec-66b67d381553

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
Production:

https://github.com/user-attachments/assets/642e4273-005b-4dee-95dd-1e1775e0a06b

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-control

**Comment by Kicu** [comment]
@dangrous @lanitochka17 I can verify that this bug is not a thing on newest main, specifically after this PR: https://github.com/Expensify/App/pull/51555 which is merged. It also introduces unit tests, so as part of this issue we can add a test case with just the `,`.

I can confirm that this error appears right now both in staging and in prod, but on newest `main` it should not be there.

**Comment by Kicu** [comment]
### my local dev
https://github.com/user-attachments/assets/b278af76-0b82-4f12-b68a-ce2ef5123246

**Comment by dangrous** [comment]
hmm @Kicu I still see the error on `main`, and also I don't see it on `production`... Can you check again? If it's truly fixed on main (and/or production) we can move this through as not a blocker since it's an edge case, but that's not what I'm seeing with my testing. Thanks!

And it's definitely from https://github.com/Expensify/App/pull/50883 (https://github.com/Expensify/App/commit/d3f99c199aa24a6949749ca9974aacf65bfe7772)

The error is:

```
SearchQueryUtils.ts:223 Error when parsing SearchQuery: "," SyntaxError: Expected default key, end of input, key, quote, whitespace, or word but "," found. (at searchParser.js:20:1)
    at new peg$SyntaxError (searchParser.js:20:1)
    at peg$buildStructuredError (searchParser.js:415:1)
    at Module.peg$parse (searchParser.js:1195:1)
    at Module.buildSearchQueryJSON (SearchQueryUtils.ts:214:20)
    at eval (SearchRouter.tsx:118:23)
    at invokeFunc (debounce.js:102:19)
    at trailingEdge (debounce.js:151:14)
    at timerExpired (debounce.js:139:14)
```

**Comment by luacmartins** [comment]
I also see it on main and agree it comes from https://github.com/Expensify/App/pull/50883. That being said, I don't think this is a blocker since it only breaks the syntax in a few edge cases

**Comment by dangrous** [comment]
yeah i think that's fine - I'll switch to daily. Do you want to take over? I can also continue to shepherd, whichever you prefer.

**Comment by luacmartins** [comment]
I can take over

**Comment by Kicu** [comment]
I think I was imprecise in my previous message 😅 

What I believe is not a bug right now is just typing the `,` in the input. If you type it, we will parse it with the new/lighter autocomplete parser which now (after #51555) correctly handles a single comma query.
That is the error that was presented in the original video in the issue and that is something that I verified on main that is not a bug anymore. It is a bug on staging - you can check yourself, open devtools, open Router, type a `,` and see the error as in the video.

However I can see that the error @dangrous described here: https://github.com/Expensify/App/issues/51661#issuecomment-2444463214 is **after** typing `,` and then hitting ENTER I think.
I'm believe that's the case, because I see in the stack trace: `(searchParser.js:20:1)` which is the main/heavier parser.

**tl;dr:**
If you just type a character we parse it with a different parser and it works. But if you type and then submit/hit enter we get the error.

@289Adam289 if you find some time you can take a look, perhaps we should allow a single comma in the main parser as well 🤔

**Comment by 289Adam289** [comment]
Sure we can allow commas in the main parser. I will work on that.

**Comment by luacmartins** [comment]
This is done and no payment is needed here.
